snmalloc.yaml

```diff
@@ -11,7 +11,7 @@ package:
 
 vars:
   # https://github.com/microsoft/snmalloc/issues/669
-  llvm-vers: 18
+  llvm-vers: 17
```

Comment:
I'm no LLVM expert, but this looks like a downgrade and I'm curious why this was needed. If this value matters, can we have a comment explaining why?

Reply:
@luhring lu No expert on clang or llvm either, with version 18 we started getting an error `redefinition of a 'extern inline' function 'memcpy' is not supported in C++`

Cause according to claude AI
```
Root Cause
- LLVM 18 is more strict about attribute placement and function redefinitions
- The issue appears to be with memcpy redefinition in snmalloc/override/memcpy.cc
- The SNMALLOC_EXPORT attribute is being applied after a previous definition
```

it suggested me to do some code changes. I started looking into [upstream guide](https://github.com/microsoft/snmalloc/blob/main/docs/BUILDING.md/#building-on-unix-like-platforms) and it suggested `dependencies: CMake, Ninja, Clang 6.0 or later and a C++17 standard library.`

I try to downgrade the version to suggested 17 and it worked.

Reply:
Gotcha! Can we add a comment that summarizes that reason for staying on 17?

Reply:
Comment added.

Reply:
Sorry — to the melange file — so that the next person working on this package has the context!

Reply:
Opps 😄